Menus/ScenarioGUI.cpp

```diff
@@ -26,2022 +12,1051 @@
-            hline(drawBitmap, sitePos.m_X - circleRadius - 1, sitePos.m_Y + circleRadius + 1, sitePos.m_X + circleRadius, color);
-            vline(drawBitmap, sitePos.m_X - circleRadius - 1, sitePos.m_Y - circleRadius - 1, sitePos.m_Y + circleRadius, color);
-            vline(drawBitmap, sitePos.m_X - circleRadius - 1 - 1, sitePos.m_Y - circleRadius - 1, sitePos.m_Y + circleRadius, color);
-            vline(drawBitmap, sitePos.m_X + circleRadius, sitePos.m_Y + circleRadius, sitePos.m_Y - circleRadius - 1, color);
-            vline(drawBitmap, sitePos.m_X + circleRadius + 1, sitePos.m_Y + circleRadius, sitePos.m_Y - circleRadius - 1, color);
-        }
-        else
-        {
-            circle(drawBitmap, sitePos.m_X, sitePos.m_Y, circleRadius, color);
-            circle(drawBitmap, sitePos.m_X, sitePos.m_Y, circleRadius - 1, color);            
-        }
-    }
-
-    return totalSegments <= onlyFirstSegments && totalSegments <= onlyLastSegments;
-}
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void ScenarioGUI::DrawLineToSitePoint(BITMAP *drawBitmap) const {
+		int drawColor = c_GUIColorWhite;
+		int blendAmount = 0;
+
+		for (int index = 1; index < m_LineToSitePoints.size(); index++) {
+			int lineStartX = m_LineToSitePoints.at(index - 1).GetFloorIntX();
+			int lineStartY = m_LineToSitePoints.at(index - 1).GetFloorIntY();
+			int lineEndX = m_LineToSitePoints.at(index).GetFloorIntX();
+			int lineEndY = m_LineToSitePoints.at(index).GetFloorIntY();
+
+			blendAmount = 195 + RandomNum(0, 30);
+			set_screen_blender(blendAmount, blendAmount, blendAmount, blendAmount);
+			line(drawBitmap, lineStartX, lineStartY, lineEndX, lineEndY, drawColor);
+
+			blendAmount = 30 + RandomNum(0, 50);
+			set_screen_blender(blendAmount, blendAmount, blendAmount, blendAmount);
+			line(drawBitmap, lineStartX + 1, lineStartY, lineEndX + 1, lineEndY, drawColor);
+			line(drawBitmap, lineStartX - 1, lineStartY, lineEndX - 1, lineEndY, drawColor);
+			line(drawBitmap, lineStartX, lineStartY + 1, lineEndX, lineEndY + 1, drawColor);
+			line(drawBitmap, lineStartX, lineStartY - 1, lineEndX, lineEndY - 1, drawColor);
+		}
+
+		// Draw a circle around the selected site point.
+		Vector sitePos = m_PlanetCenter + m_SelectedScene->GetLocation() + m_SelectedScene->GetLocationOffset();
+		int sitePosX = sitePos.GetFloorIntX();
+		int sitePosY = sitePos.GetFloorIntY();
+		int circleRadius = 8;
+
+		blendAmount = 195 + RandomNum(0, 30);
+		set_screen_blender(blendAmount, blendAmount, blendAmount, blendAmount);
+		circle(drawBitmap, sitePosX, sitePosY, circleRadius, drawColor);
+
+		blendAmount = 120 + RandomNum(0, 50);
+		set_screen_blender(blendAmount, blendAmount, blendAmount, blendAmount);
+		circle(drawBitmap, sitePosX, sitePosY, circleRadius - 1, drawColor);
```

Comment:
Is there no way to draw a circle with 2 pixels of border width? Seems to me this could be compacted a lot if we could do that - wouldn't need sitePosX/Y or circleRadius, and as mentioned above, the blending seems to do basically nothing.